compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/ConversionUtils.kt

```diff
@@ -204,6 +212,15 @@ private fun FirConstKind<*>.toIrConstKind(): IrConstKind<*> = when (this) {
     FirConstKind.IntegerLiteral, FirConstKind.UnsignedIntegerLiteral -> throw IllegalArgumentException()
 }
 
+internal fun FirTypeRef.toFirClassSymbolOrNull(session: FirSession): FirClassSymbol<*>? {
+    if (this is FirResolvedTypeRef) {
+        if (type is ConeClassLikeType) {
+            return (type as ConeClassLikeType).lookupTag.toSymbol(session).safeAs()
```

Comment:
You can use `coneTypeSafe` for such transformation. We will think about better utility functions. At this moment I inlined this function.

Reply:
Good to learn that, thanks!